### ISSUE TITLE
Bug Report/Feature Request: Can't manipulate value of searchable Northstar-Dropdown 

### ISSUE THREAD

**Issue by marwalsch** [microsoft/fluentui#16814]
When a [Dropdown](https://fluentsite.z22.web.core.windows.net/0.52.0/components/dropdown/definition) is searchable there seems to be no way to manually manipulate the value of the input field. 
```
                <Dropdown
                    search
                    clearable={false}
                    items={["hello", "bye"]}
                    value={this.state.value}  //Or any static value
                />
```
No passed Value is ever displayed inside the dropdown. You can check by making it clearable; The dropdown is empty but displays the X-Button, which can be pressed to no avail.

A Fix might already inlcude this, but it might be really useful to actually manipulate and read from the current search-input of the dropdown, such as when clearing it manually through an event triggered in the parent component.

**Comment by joschect** [comment]
@marwalsch Hi! Thanks for the submission. I'm a little unclear as to what you mean by "manually manipulate" the dropdown. I believe that the V0 dropdown allows a value to be passed in and for it to be controlled. I made this [codesandbox here](https://codesandbox.io/s/fluent-ui-example-forked-3lrw2?file=/example.js) and I was able to set that value. The value is not changing because I don't have onchange hooked up.

**Comment by marwalsch** [comment]
@joschect Thanks for your reply. Please refer to this [updated example](https://codesandbox.io/s/fluent-ui-example-forked-rvxpr?file=/example.js) where the dropdown is searchable and clearable. You will notice that no value is set and clearing has no effect, effectively disabling the dropdown. 
As for my latter remark, I am in a situation where I need to enable and disable the dropdown based on an event. however, due to this bug, I cannot clear the dropdown (=set value to "") when re-enabling it.

**Comment by yStreahn** [comment]
@marwalsch Hey! If you want to manipulate the displayed value of the dropdown a.k.a the current search you need to change the "searchQuery" property.

**Comment by marwalsch** [comment]
@yStreahn Thanks, this allows to initially set the value, but seems to be unable to be manipulated like regular input fields? Doing as follows doesn't allow you to edit the initial value [Sandbox](https://codesandbox.io/s/fluent-ui-example-forked-rvxpr?file=/example.js).
```

    handleSearchChange = (e: any, data: any) => {
        this.setState({
            value: data.value
        });
    };

                   <Dropdown
                        fluid
                        search
                        clearable={true}
                        searchQuery={this.state.value}
                        onChange={this.handleSearchChange}
                    />
```

**Comment by yStreahn** [comment]
@marwalsch Exactly. The onChange is only triggered when the value changes (item is selected, cleared). There is a separate change handler called onSearchQueryChange. But it somehow does not get called when the dropdown is cleared so you would need to trigger it manually in the onChange event.

**Comment by marwalsch** [comment]
@yStreahn It appears simply [exchanging the `onChange `with `onSearchQueryChange](https://codesandbox.io/s/fluent-ui-example-forked-rvxpr?file=/example.js) `doesn't work as expected, as the passed data value is frequently null rather than the actual input.

**Comment by yStreahn** [comment]
@marwalsch Ah yeah so your onSearchQueryChange should look something like this.
`handleSearchChange = (e, data) => {
    this.setState({
      value: data.searchQuery
    });
  };`

**Comment by marwalsch** [comment]
@yStreahn Thanks a lot, that made the selection work. 
I noticed two things though:

1. `onSelect `is always called when the search-field is typed in and, making it somewhat unnecessary compared to the `onSearchQueryChange`. I imagine this behaviour to be unintentional (?).
2. As pointed out earlier, clearing the dropdown does not actually do what it's supposed to do now ([Issue #12314](https://github.com/microsoft/fluentui/issues/12314)). Is there any way to get around this? (other than adding a button that does setState(value:""), as I'd like to keep the inline X-Button if possible.

**Comment by yStreahn** [comment]
@marwalsch You can do something like this

`handleSearchChange = (e, data) => {
    this.setState({
      value: data.value ? data.value : ""
    });
  };

  handleChange = (e, data) => {
    if (data.value == null) {
      this.handleSearchChange(e, data);
    } else {
      //do stuff
    }
  };

  //@Override
  render() {
    return (
      <Dropdown
        search
        clearable={true}
        items={["1st option", "2nd option"]}
        searchQuery={this.state.value}
        onSearchQueryChange={this.handleSearchChange}
        onChange={this.handleChange}
      />
    );
  }`

**Comment by marwalsch** [comment]
@yStreahn I did something similar. Currently the behaviour is calling `onSelect`   when the user explicitly selects an searched item, which is then mapped to an object. But since this is called every time, it checks the map each time the query changes, even though i can assume it's 99% not matching gibberish when a term is entered. This definitely works but might be a performance concern.

**Comment by gouttierre** [comment]
@marwalsch - thanks for filing this initially. I see there has been some feedback and progress with the suggested workaround. Regarding your performance concern - we will have to review it internally with the team. I will mark this one to be be reviewed.
